docs/commerce/shipping_management/extend_shipping.md

```diff
@@ -0,0 +1,155 @@
+---
+description: Extend Shipping with custom shipping method type and other extra features.
+edition: commerce
+---
+
+# Extend shipping
+
+There are different ways in which you can extend or customize your Shipping module implementation. 
+Here, you can learn about the following ideas to make your Commerce solution more powerful:
+
+- create a custom shipping method type
+- toggle shipping method availability in checkout based on a condition
+- display shipping method parameters on the shipping method details page
+
+You can also [customize the shipment processing workflow](configure_shipment.md#custom-shipment-workflows).
+
+## Create custom shipping method type
+
+If your application needs shipping methods of other type than the default ones, you create custom shipping method types. 
+See the code samples below to learn how you could do it.
+ 
+### Define custom shipping method type class
+
+Create a definition of the shipping method type. 
+Use a built-in type factory to define the class in `config/services.yaml`:
+
+``` yaml
+[[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 0, 1) =]] [[= include_file('code_samples/front/shop/shipping/config/packages/services.yaml', 1, 8) =]]
+```
+
+At this point a custom shipping method type should be visible in the user interface.
+
+### Create options form
+
+In `src/ShippingMethodType/Form/Type/`, create the `CustomShippingMethodOptionsType.php` form type that corresponds to the custom shipping method type.
+
+You also define a name of the custom shipping method type here, by using the `getTranslationMessages` method.
+
+``` php hl_lines="32"
+[[= include_file('code_samples/front/shop/shipping/src/ShippingMethodType/Form/Type/CustomShippingMethodOptionsType.php') =]]
+```
+
+In `translations/`, create a translations file that stores a name value for the custom shipping method type, for example:
```

Comment:
It could be interesting to give the file name as reader may not remember the `<domain>.<lang>.<format>` convention.
```suggestion
In `translations/`, create a translations file `ibexa_shipping.en.yaml` that stores a name value for the custom shipping method type, for example:
```